sw/device/lib/dif/dif_padctrl.h

```diff
@@ -0,0 +1,157 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_PADCTRL_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_PADCTRL_H_
+
+#include <stdbool.h>
+
+// Generated.
+#include "padctrl_regs.h"
+
+#include "sw/device/lib/base/mmio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Pad attributes.
+ *
+ * Some attributes have defined meanings, others are implementation defined
+ * (`kDifPadCtrlAttrAdditionalFirst` and onwards). Not all pads will support all
+ * attributes.
+ */
+typedef enum dif_padctrl_attr {
```

Comment:
I don't remember whether it has already been discussed, but is it not feasible to define allowed and disallowed combinations in the enum, like @alphan did in GPIO DIF? That would eliminate some added complexity in the source file.

Reply:
With padctrl we cannot know this ahead of time, due to the additional attributes (which we need to include in the interface). We've had this discussion in a few places (because unlike GPIO, choosing conflicting attributes could damage the chip #1831), and garret and I have made a decision about how we would like this interface to look.

Reply:
> because unlike GPIO, choosing conflicting attributes could damage the chip

This is a separate problem.

> With padctrl we cannot know this ahead of time, due to the additional attributes

Thats fair enough

Reply:
@silvestrst and I discussed this, and I also discussed this with garret.

If we move to a mask based interface, then known good attribute combinations can be provided by a chip-specific file (like top_earlgrey, but associated more specifically with a silicon chip implementation). It's not the responsibility of the DIF to provide these known good masks.

Reply:
I have switched to the mask based interface but ignored additional/custom attributes for now. One way to support these would be to define a constant `kDifPadctrlAttrAdditionalFirstShift` or similar which would be used to generate the shift amount `dif_padctrl_attr_t second_additional = 1 << (kDifPadctrlAttrAdditionalFirstShift + 1)`. Though I'll end up declaring the constant here as an extern than defining it within the .c (as we don't want `static` items in DIF headers). I haven't done this for now to leave the topic open for discussion.